### PR TITLE
Fix and unskip test_participant_selection

### PR DESCRIPTION
**Pull request by andrevmatos**
By accounting for the possibility of the settled channel be already cleaned. Also, improve determinism by serializing nodes joining the network.
May also depend on #2006 to fully fix the test.

### CODE REVIEW

File: raiden/tests/integration/long_running/flaky/test_token_networks.py [hunk 1]
```diff
@@ -84,25 +83,14 @@ def test_participant_selection(raiden_network, token_addresses, skip_if_tester):
...
     # connect the first node (will register the token if necessary)
-    RaidenAPI(raiden_network[0].raiden).token_network_connect(
-        registry_address,
-        token_address,
-        100,
-    )
-
-    # connect the other nodes
-    connect_greenlets = [
-        gevent.spawn(
```
Comment (hackaugusto) [inline raiden/tests/integration/long_running/flaky/test_token_networks.py]: why remove the parallel registration?

File: raiden/tests/integration/long_running/flaky/test_token_networks.py [hunk 2]
```diff
@@ -231,4 +219,5 @@ def test_participant_selection(raiden_network, token_addresses, skip_if_tester):
         token_address=token_address,
         partner_address=sender.address,
     )
-    assert receiver_channel[0].settle_transaction is not None
+    # because of timing, channel may already have been cleaned
```
Comment (hackaugusto) [inline raiden/tests/integration/long_running/flaky/test_token_networks.py]: You can use `waiting.wait_for_settle` here and then the settle transaction must be set.

Btw, the channel will not be cleared after settle, but after the unlocks.

File: raiden/connection_manager.py [hunk 3]
```diff
@@ -265,39 +269,84 @@ def _open_channels(self):
...
-                # partner opens first.
-                log.info('partner opened channel first')
-
-            try:
-                self.api.set_total_channel_deposit(
-                    self.registry_address,
-                    self.token_address,
-                    partner,
-                    self._initial_funding_per_partner,
+        while True:
```
Comment (pcppcp) [inline raiden/connection_manager.py]: just a nitpick: could this be split into smaller functions?

File: raiden/connection_manager.py [hunk 4]
```diff
@@ -265,39 +269,84 @@ def _open_channels(self):
...
+                except AddressWithoutCode:
+                    log.warn('connection manager: channel closed just after it was created')
+                except TransactionThrew:
+                    log.exception('connection manager: deposit failed')
+                except ChannelIncorrectStateError:
+                    log.exception('connection manager: channel not in opened state')
+                else:
+                    return True
+
+            greenlets = [
```
Comment (pcppcp) [inline raiden/connection_manager.py]: what if the gevent fails? will the exception propagate to the central hub? 
How about checking return value of `_join_partner()` and logging at least a warning if something went wrong?

File: raiden/connection_manager.py (hunk 3 above)
Comment (andrevmatos) [reply raiden/connection_manager.py]: Ok, split `_join_partner` to its own method, but as the other parts of this function are kind of chained together, I couldn't get an easy way to split them. I think it's small enough now.

File: raiden/connection_manager.py (hunk 4 above)
Comment (andrevmatos) [reply raiden/connection_manager.py]: They will propagate up the stack. These exceptions handled there are the only ones that may happen sometimes, and are harmless, as it's run in a loop.

File: raiden/connection_manager.py [hunk 5]
```diff
@@ -252,52 +246,91 @@ def find_new_partners(self, number: int):
...
+                self.token_address,
+                partner,
+                self._initial_funding_per_partner,
+            )
+        except TransactionThrew:
+            log.exception('connection manager: deposit failed')
+        except ChannelIncorrectStateError:
+            log.exception('connection manager: channel not in opened state')
+
     def _open_channels(self):
```
Comment (hackaugusto) [inline raiden/connection_manager.py]: this method should be renamed

File: raiden/tests/integration/long_running/test_token_networks.py [hunk 6]
```diff
@@ -35,8 +35,7 @@ def wait_for_transaction(
 def is_channel_open_and_funded(channel_state):
     return (
         channel.get_status(channel_state) == CHANNEL_STATE_OPENED and
-        channel_state.our_state.contract_balance > 0 and
-        channel_state.partner_state.contract_balance > 0
+        channel_state.our_state.contract_balance > 0
```
Comment (hackaugusto) [inline raiden/tests/integration/long_running/test_token_networks.py]: why did you remove the other test?

File: raiden/tests/integration/long_running/test_token_networks.py [hunk 7]
```diff
@@ -49,6 +48,7 @@ def is_manager_saturated(connection_manager, registry_address, token_address):
         lambda channel_state:
         (
             is_channel_open_and_funded(channel_state) and
+            channel_state.partner_state.address != connection_manager.BOOTSTRAP_ADDR and
```
Comment (hackaugusto) [inline raiden/tests/integration/long_running/test_token_networks.py]: could you please move this to a normal function? I would really prefer if we stick to single line lambdas

File: raiden/tests/integration/long_running/test_token_networks.py [hunk 8]
```diff
@@ -154,10 +153,26 @@ def test_participant_selection(raiden_network, token_addresses, skip_if_tester):
             assert routes is not None
 
     # create a transfer to the leaving node, so we have a channel to settle
-    sender = raiden_network[-1].raiden
-    receiver = raiden_network[0].raiden
-
+    for app in raiden_network:
+        sender = app.raiden
+        sender_channel = next((
```
Comment (hackaugusto) [inline raiden/tests/integration/long_running/test_token_networks.py]: can you please simplify this code?

File: raiden/tests/integration/long_running/test_token_networks.py [hunk 9]
```diff
@@ -154,10 +153,26 @@ def test_participant_selection(raiden_network, token_addresses, skip_if_tester):
...
+            if channel_state.our_state.contract_balance > 0 and
+            channel_state.partner_state.contract_balance > 0
+        ), None)  # choose a fully funded channel from sender
+        if sender_channel:
+            break
     registry_address = sender.default_registry.address
+
+    receiver = next(
+        app.raiden for app in raiden_network
+        if app.raiden.address == sender_channel.partner_state.address
```
Comment (hackaugusto) [inline raiden/tests/integration/long_running/test_token_networks.py]: what if the loop above is exhausted and not `sender_channel` is set?

File: raiden/tests/integration/long_running/test_token_networks.py (hunk 6 above)
Comment (andrevmatos) [reply raiden/tests/integration/long_running/test_token_networks.py]: It required that all channels we have would be counter-deposited, but this may not be the case when we connected to a client already saturated. This will just assert our side of our channels were funded.

File: raiden/tests/integration/long_running/test_token_networks.py (hunk 9 above)
Comment (andrevmatos) [reply raiden/tests/integration/long_running/test_token_networks.py]: This never happens, as there **will** be at least 2 nodes with **some** channel which got counter-deposited, per the rules on `ConnectionManager._open_channels`.
